compiler/crates/relay-lsp/src/utils.rs

```diff
@@ -116,14 +117,31 @@ pub fn get_project_name_from_file_group(file_group: &FileGroup) -> Result<String
 /// request, only if the request occurs within a GraphQL document or Docblock.
 pub fn extract_feature_from_text(
     project_config: &ProjectConfig,
-    source_feature_cache: &DashMap<Url, Vec<JavaScriptSourceFeature>>,
+    js_source_feature_cache: &DashMap<Url, Vec<JavaScriptSourceFeature>>,
+    schema_source_cache: &DashMap<Url, GraphQLSource>,
     text_document_position: &TextDocumentPositionParams,
     index_offset: usize,
 ) -> LSPRuntimeResult<(Feature, Span)> {
     let uri = &text_document_position.text_document.uri;
     let position = text_document_position.position;
 
-    let source_features = source_feature_cache
+    if let Some(schema_source) = schema_source_cache.get(uri) {
+        let source_location_key = SourceLocationKey::standalone(uri.as_ref());
+        let schema_document = graphql_syntax::parse_schema_document(
```

Comment:
@captbaritone I recreated the behavior of JavaScript sources, where the TextSource is kept in memory and for each feature extraction the parsing process is run. In the case of the schema the parsing might be more expensive (I imagine it will be on Facebook's schema at least 😅).
Would you recommend a different strategy or do you think this is acceptable?

Reply:
At this point our main schema is not actually written to disk any more, so I don't think that will be a problem.